src/index/TextIndexLiteralFilter.cpp

```diff
@@ -0,0 +1,57 @@
+// Copyright 2025, University of Freiburg,
+// Chair of Algorithms and Data Structures.
+// Author: Felix Meisen (fesemeisen@outlook.de)
+
+#include "index/TextIndexLiteralFilter.h"
+
+#include <re2/re2.h>
+
+// _____________________________________________________________________________
+TextIndexLiteralFilter::TextIndexLiteralFilter()
+    : regex_{std::make_unique<RE2>("(?s).*")},
+      filterType_{LiteralFilterType::AcceptMatching},
+      addAllLiterals_(true){};
+
+// _____________________________________________________________________________
+TextIndexLiteralFilter::TextIndexLiteralFilter(
+    const TextIndexLiteralConfiguration& config)
+    : regex_{std::make_unique<RE2>(config.predicateRegex_, RE2::Quiet)},
+      filterType_{config.isWhitelistOrBlacklist_},
+      addAllLiterals_(config.addAllLiterals_) {
+  if (!regex_->ok()) {
+    throw std::runtime_error{
+        absl::StrCat("The regex supposed to filter predicates for which the "
+                     "objects are stored in the text index was \"",
+                     regex_->pattern(),
+                     "\". This is not supported by QLever (which uses "
+                     "Google's RE2 library); "
+                     "the error from RE2 is: ",
+                     regex_->error())};
+  }
+}
+
+// _____________________________________________________________________________
+std::tuple<bool, bool, bool> TextIndexLiteralFilter::computeInTextIndexMap(
+    const TripleComponent& s, const TripleComponent& p,
+    const TripleComponent& o) const {
+  // If add all Literals check if they are a literal
+  if (addAllLiterals_) {
+    return {s.isLiteral(), p.isLiteral(), o.isLiteral()};
```

Comment:
If this makes your E2E tests fail if you return `false` for the subject, then probably something is wrong with the vocabulary merger (we add some subject literals internally, but those also appear as objects in other triples, so they should be added to the text index, because we have "or" semantics in the merger, don't we?